test/core/event_engine/test_suite/README.md

```diff
@@ -37,3 +37,28 @@ you could depend on any subset of the following:
 * `//test/core/event_engine/test_suite:dns`
 * `//test/core/event_engine/test_suite:client`
 * `//test/core/event_engine/test_suite:server`
+
+
+## Breaking Changes
+
+As the conformance test suite evolves, it may introduce changes that break your
+implementation. We want to be able to improve the conformance test suite freely
+without harming everyone's EventEngine builds or requiring lock-step engine
+improvements, so a test rollout strategy is required.
+
+New conformance tests will be added to `*_rc` (release candidate) targets,
+such as:
+
+* `//test/core/event_engine/test_suite:timer_rc`
+* `//test/core/event_engine/test_suite:dns_rc`
+* `//test/core/event_engine/test_suite:client_rc`
+* `//test/core/event_engine/test_suite:server_rc`
+
+Quarterly, these additional tests will be rolled up into the main test suite
```

Comment:
as discussed offline:
- let's define this in terms of releases
- we'll probably want some kind of automation to make this less painful; perhaps instead of suffixing not-yet-required tests with _rc we can go the other way and add a suffix to the tests that are required. That makes it easier to discover the correct place to add test cases.

Reply:
@tamird  Regarding the naming scheme, I'm interested in what you think a clearer scheme might be. It's a common standard, for example, to have a production version `<name>` and a release candidate version named `<name>-rc1` or some such. gRPC versions its libraries that way. What I suggested is the same, but for bazel targets.

Given that a new test may be added to the `_rc` target the day before a release, it may be better to have a plan such as:

* "base name" targets make up the current production conformance test suite (e.g., `timer`, `dns`, etc)
* new tests get added to brand new independent targets (e.g., `timer_dst_expectations`)
* targets get tagged with a release version or a date that they're expected to land in production, and the production target name that they should be rolled up into (e.g., `timer`).
* release automation will combine the targets that need to be rolled up (via adding its source and headers files, maybe), and delete the old independent targets

Reply:
> Given that a new test may be added to the _rc target the day before a release, it may be better to have a plan such as:

The wall clock time between the test being added and its inclusion in a release is immaterial in my thinking. What matters is that adding new conformance tests is treated as an API change and thus doesn't happen in minor releases. Does that make sense?

Reply:
I see. The change would have to be scheduled for minor releases. It's a bit of a sticky subject, but given that this low-level API is expected to be sparsely used in the public and primarily by "power users", after the EventEngine API is baked for a while and promoted from experimental, we intend to provide best effort API stability without a major version bump for necessary changes. And we're aiming to design the API to allow some changes to be made in a backwards-compatible ways.

Last I talked with the team about it, there is strong sentiment against doing a gRPC 2.0 (has been for years, anyway), so your suggestion would prevent us from improving the test suite at all. We do a minor release every 6 weeks, with a pre-release a week before. You can read more about the release process here: https://grpc.io/docs/what-is-grpc/faq/#how-long-are-grpc-releases-supported-for

Reply:
As we discussed offline:

* minor gRPC releases (every 6 weeks) will see new updates added to the conformance test suite.
* customers that live at HEAD will see new conformance tests immediately as they are merged.
* ideally, EventEngine implementers will have a way to "assert failure" of specified subsets of the conformance tests. These tests will still run, asserting failure instead of success. This will allow most tests to continue to run for a given implementation, while asserting failure of some tests for known engine deficiencies. Implementers will still be protected from regressions on the subset of conformance tests they do not opt out of, and their builds will fail if the Engine implementation is fixed for some test they've "asserted failure" on.

Once I've figured out how bazel might allow us to do this, I'll update the docs accordingly.